api/views.py

```diff
@@ -147,6 +169,282 @@ def get_queryset(self):
+        return RealPhone.objects.filter(user=self.request.user)
+
+
+    def create(self, request):
+        """
+        Add real phone number to the authenticated user.
+
+        The "flow" to verify a real phone number is:
+        1. POST a number (Will text a verification code to the number)
+        2a. PATCH the verification code to the realphone/{id} endpoint
+        2b. POST the number and verification code together
+
+        The authenticated user must have a subscription that grants one of the
+        `SUBSCRIPTIONS_WITH_PHONE` capabilities.
+
+        The `number` field should be in [E.164][e164] format which includes a country
+        code. If the number is not in E.164 format, this endpoint will try to
+        create an E.164 number by prepending the country code of the client
+        making the request (i.e., from the `X-Client-Region` HTTP header).
+
+        If the `POST` does NOT include a `verification_code` and the number is
+        a valid (currently, US-based) number, this endpoint will text a
+        verification code to the number.
+
+        If the `POST` DOES include a `verification_code`, and the code matches
+        a code already sent to the number, this endpoint will set `verified` to
+        `True` for this number.
+
+        [e164]: https://en.wikipedia.org/wiki/E.164
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        # First, check if the request includes a valid verification_code
+        # value, look for any un-expired record that matches both the phone
+        # number and verification code and mark it verified.
+        verification_code = serializer.validated_data.get(
+            "verification_code"
+        )
+        if verification_code:
+            valid_record = get_valid_realphone_verification_record(
+                request.user,
+                serializer.validated_data["number"],
+                verification_code
+            )
+            if not valid_record:
+                raise exceptions.ValidationError("Could not find that verification_code for user and number. It may have expired.")
+
+            headers = self.get_success_headers(serializer.validated_data)
+            verified_valid_record = valid_record.mark_verified()
+            response_data = model_to_dict(verified_valid_record, fields=[
+                "id", "number", "verification_sent_date", "verified",
+                "verified_date"
+            ])
+            return response.Response(
+                response_data, status=201, headers=headers
+            )
+
+        # We call an additional _validate_number function with the request
+        # to try to parse the number as a local national number in the
+        # request.country attribute
+        valid_number = _validate_number(request)
+        serializer.validated_data["number"] = valid_number.phone_number
+
+
+        self.perform_create(serializer)
```

Comment:
Should we detect if a `RealPhone` has a still-valid verification code and skip creating and sending a new one? I could see this being used to:

1) Send 1,000s of SMS messages to a target phone, to annoy them
2) Generate 10,000 verification codes, to increase the chances of brute forcing